graphql/database/base/DynamoDBModel.js

```diff
@@ -0,0 +1,213 @@
+
+    // Add any secondary indexes.
+    // https://github.com/clarkie/dynogels#global-indexes
+    if (this.indexes) {
+      options.indexes = this.indexes
+    }
+
+    this.dynogelsModel = dynogels.define(this.name, options)
+  }
+
+  /**
+   * Any secondary indexes on the model.
+   * See:
+   *   https://github.com/clarkie/dynogels#global-indexes
+   *   https://github.com/clarkie/dynogels#local-secondary-indexes
+   * @return {object} The name of the hashKey for the DynamoDB table.
+   * @return {array<object>} indexes<index> - A list of index objects.
+   * @return {string} index.hashKey - The hash key.
+   * @return {string} index.rangeKey - The range key.
+   * @return {string} index.name - The name of the index.
+   * @return {string} index.type - Either "global" or "local".
+   */
+  static get indexes() {
+    return null
+  }
+
+  /**
+   * The permissions object, used to check authorization for database
+   * operations. By default, no operations are authorized.
+   * @return {object} The permissions object, with a key for each
+   *   operation name. Each property value is a function that receives:
+   *    - a userContext object
+   *    - item hashKey (if getting a specific item)
+   *    - item rangeKey (if one exists)
+   *    - item object (if creating or updating an item)
+   *   The authorizer function returns a boolean for whether the query is authorized.
+   *   Secondary indexes must be authorized separately in `indexPermissions`,
+   *   with a property key set to the name of the secondary index.
+   */
+  static get permissions() {
+    return {
+      // Receives: userContext, hashKeyValue, rangeKeyValue
+      get: () => false,
+      // Receives: userContext
+      getAll: () => false,
+      // Receives: userContext, hashKeyValue, rangeKeyValue, item
+      update: () => false,
+      // Receives: userContext, hashKeyValue, rangeKeyValue, item
+      create: () => false,
+      indexPermissions: {},
+    }
+  }
+
+  static async get(userContext, hashKey, rangeKey) {
+    const keys = [hashKey]
+    if (rangeKey) {
+      keys.push(rangeKey)
+    }
+    // logger.debug(`Getting obj with hashKey ${hashKey} from table ${this.tableName}.`)
+    if (!this.isQueryAuthorized(userContext, 'get', hashKey, rangeKey)) {
+      throw new UnauthorizedQueryException()
+    }
+    try {
+      const data = await this.dynogelsModel.getAsync(...keys)
+      if (isNil(data)) {
+        throw new DatabaseItemDoesNotExistException()
+      }
+      return this.deserialize(data)
+    } catch (e) {
+      throw e
+    }
+  }
+
+  static async getInternal(keys) {
+    return this.dynogelsModel.getAsync(...keys)
+  }
+
+  static async getBatchInternal(keys, options) {
+    return this.dynogelsModel.getItemsAsync(keys, options)
+  }
+
+  static async getAllInternal() {
+    const items = await this.dynogelsModel
+      .scan()
+      .loadAll()
+      .execAsync()
+    return items.Items
+  }
+
+  static query(userContext, hashKey) {
```

Comment:
Should this be here? I assume the "connector" should only implement the internal methods?

Reply:
The query returns a `queryObj` that additional filters can be applied to, which wouldn't be applicable to all `Model` instances.